lib/src/git.rs

```diff
@@ -1275,50 +1275,30 @@ pub fn push_updates(
     updates: &[GitRefUpdate],
     callbacks: RemoteCallbacks<'_>,
 ) -> Result<(), GitPushError> {
-    let mut temp_refs = vec![];
     let mut qualified_remote_refs = vec![];
     let mut refspecs = vec![];
     for update in updates {
         qualified_remote_refs.push(update.qualified_name.as_str());
         if let Some(new_target) = &update.new_target {
-            // Create a temporary ref to work around https://github.com/libgit2/libgit2/issues/3178
-            let temp_ref_name = format!("refs/jj/git-push/{}", new_target.hex());
-            temp_refs.push(git_repo.reference(
-                &temp_ref_name,
-                git2::Oid::from_bytes(new_target.as_bytes()).unwrap(),
-                true,
-                "temporary reference for git push",
-            )?);
             refspecs.push(format!(
                 "{}{}:{}",
                 (if update.force { "+" } else { "" }),
-                temp_ref_name,
+                git2::Oid::from_bytes(new_target.as_bytes()).unwrap(),
                 update.qualified_name
             ));
         } else {
             refspecs.push(format!(":{}", update.qualified_name));
         }
     }
-    let result = push_refs(
+    // Please don't inline `push_refs` before 2024-06-01 without talking to ilyagr,
+    // he's working on this code.
```

Comment:
??

Could you add some more context? Is it related to force-with-lease?

Reply:
Yes. If this really bothers you, I can online the function right now, but I likely want to extract most of push_refs as a separate function.

I'll link to a draft PR in a bit.

Reply:
See #3522

Reply:
I might also just remove the comment and merge this. 
This is not particularly important.

Reply:
I'm not against leaving `push_refs()` as a separate function. I just mean the comment is unclear (e.g. why it's 2024-06-01, etc.) Something like `TODO(ilyagr): this function will be reused to implement force-with-lease, so don't inline` will help.

Reply:
I combined out two comments.